js/index.js

```diff
@@ -28,44 +27,77 @@ app.controller("IndexController", ["$scope", "$http", function($scope, $http) {
 
   function reverseGeocoding(latlng) {
     var urlString = "http://nominatim.openstreetmap.org/reverse?format=json&lat=" +
-    latlng.lat + "&lon=" +
-    latlng.lng + "&zoom=18&addressdetails=1";
-    $http.get(urlString).then(addMarker);
+      latlng.lat + "&lon=" +
+      latlng.lng + "&zoom=18&addressdetails=1";
+    $http.get(urlString).then(addMarker, errorMessage);
   }
 
   function addMarker(response) {
     var marker = {
       lat: parseFloat(response.data.lat),
       lng: parseFloat(response.data.lon),
-      message: "New entry",
+      title: "New entry",
       dueDate: new Date(),
       address: response.data.display_name
     };
-
-    $scope.markers.push(marker);
     $scope.currentMarker = marker;
     $("#modal").modal('show');
+    $("#modal").on('click', '#btnClose', function() {
+      $http.post(serverUrl, marker).then(postSuccess, errorMessage);
+      marker = {};
+    })
   }
+  $scope.remove = function(index) {
+    var id = $scope.markers[index]._id;
+    $http.delete(serverUrl + id).then(function() {
+      if ($scope.markers.length == 1) {
+        window.location.reload();
```

Comment:
We shouldn't have to do this to get the view to update, but it is a good fix for now. Maybe we can ask Oscar why the last element isn't disappearing.